src/components/posts/PostListItemCard/PostListItemCard.tsx

```diff
@@ -57,45 +69,83 @@ const useStyles = makeStyles(({ spacing, typography }) => {
+        <CardContent className={classes.cardContent}>
+          <div className={classes.leftContent}>
+            <Avatar className={classes.avatar}>
+              {props.creator.value?.avatarLetters}
+            </Avatar>
           </div>
-          <div className={classes.typeShareabilityRow}>
-            <Typography variant="h4">{props.type}</Typography>
-            <div className={classes.shareability}>
-              <Typography variant="body2">
-                <span className={classes.shareabilityLabel}>
-                  {props.shareability.includes('External')
-                    ? 'PUBLIC'
-                    : 'PRIVATE'}
-                </span>
-                {displayPostShareability(props.shareability)}
-              </Typography>
+          <div className={classes.rightContent}>
+            <div className={classes.topRightContent}>
+              <div className={classes.nameAndDate}>
+                <Typography variant="body2">
+                  {`${props.creator.value?.displayFirstName.value} ${props.creator.value?.displayLastName.value}`}
+                </Typography>
+                <Typography variant="body2" color="textSecondary">
+                  {formatDate(props.createdAt)}
+                </Typography>
+              </div>
+              {props.canEdit && (
+                <IconButton onClick={(e) => setActionsAnchor(e.currentTarget)}>
+                  <MoreVert />
+                </IconButton>
+              )}
+            </div>
+            <div className={classes.typeShareabilityRow}>
+              <Typography variant="h4">{props.type}</Typography>
+              <div className={classes.shareability}>
+                <Typography variant="body2">
+                  <span className={classes.shareabilityLabel}>
+                    {props.shareability.includes('External')
+                      ? 'PUBLIC'
+                      : 'PRIVATE'}
+                  </span>
+                  {displayPostShareability(props.shareability)}
+                </Typography>
+              </div>
             </div>
+            <Typography variant="body2">{props.body.value}</Typography>
           </div>
-          <Typography variant="body2">{props.body.value}</Typography>
-        </div>
-      </CardContent>
-    </Card>
+        </CardContent>
+      </Card>
+
+      <PostListItemMenu
+        anchorEl={actionsAnchor}
+        open={Boolean(actionsAnchor)}
+        onClose={() => setActionsAnchor(null)}
+        onEdit={() => {
+          setEdit(true);
+          setActionsAnchor(null);
+        }}
+        onDelete={() => {
+          setDeletePost(true);
+          setActionsAnchor(null);
+        }}
+      />
+      <EditPost post={props} open={edit} onClose={() => setEdit(false)} />
+      <DeletePost
+        open={deletePost}
+        project={props.project}
+        post={{
+          __typename: 'Post',
+          id: props.id,
+        }}
```

Comment:
I'm trying to avoid doing this. Makes it harder when the type changes (though this is a concrete) and harder when more info is needed, like changeset ID. 



```suggestion
        post={props}
```

Or even better
```suggestion
        post={props.post}
```
I don't think it really helps to have the post props at the root level. It makes it harder to isolate them and makes JS object identity checks harder because the object is always changing with the spread.